### DISCUSSION TITLE
Turborepo/sveltekit environment variables are not loaded in deployment, build fails

### DISCUSSION THREAD

**Discussion by isaacfink** [vercel/community#1444] [Help] [answered]
Hi guys I have the following turborepo configuration
```
{
  "$schema": "https://turbo.build/schema.json",
  "pipeline": {
    "build": {
      "dependsOn": ["^build"],
      "outputs": [".svelte-kit/**", ".vercel/**"],
      "env": ["PUBLIC_SUPABASE_URL", "PUBLIC_SUPABASE_ANON_KEY"]
    },
    "lint": {},
    "dev": {
      "cache": false,
      "persistent": true
    }
  }
}
```
It's a simple project with 2 sveltekit websites, admin and client, both websites have a supabase instance and the client is defined like this
```
import { createClient } from '@supabase/auth-helpers-sveltekit';
import { PUBLIC_SUPABASE_ANON_KEY, PUBLIC_SUPABASE_URL } from '$env/static/public';
export const client = createClient(PUBLIC_SUPABASE_URL, PUBLIC_SUPABASE_ANON_KEY);
```
in development it works perfectly but in production it fails this is the vite config 
```
import { sveltekit } from '@sveltejs/kit/vite';
import type { UserConfig } from 'vite';

const config: UserConfig = {
  plugins: [sveltekit()],
  optimizeDeps: {
    exclude: ['flowbite-svelte']
  },
  define: {
    'process.env': process.env
  }
};

export default config;
```
and the error is this:
```
admin:build:             throw new Error('supabaseUrl is required.');
admin:build:                   ^
admin:build: 
admin:build: Error: supabaseUrl is required.
admin:build:     at new SupabaseClient (/vercel/path0/node_modules/.pnpm/@supabase+supabase-js@2.4.1/node_modules/@supabase/supabase-js/dist/main/SupabaseClient.js:55:19)
```
The issue is that the environment variables are not being loaded in both I don't understand why, I made sure the variables are configured correctly on vercel

I don't know if it's a vite, sveltekit, turborepo or vercel issue

**Comment by amyegan** [comment, marked as answer]
I don't know which version of SvelteKit you're using, but this sounds like it's related to [prefixed environment variables](https://vercel.com/docs/concepts/projects/environment-variables#:~:text=The%20following-,prefixed%20Environment%20Variables,-will%20be%20available). Since you're using Vite, you may need to use the `VITE_VERCEL_` prefix instead. 

If you're using the latest version of SvelteKit, do you have a [`svelte.config.js`](https://kit.svelte.dev/docs/configuration) defined for your project? If so, make sure that [`publicPrefix`](https://kit.svelte.dev/docs/configuration#env) isn't set to something other than the default value (`"PUBLIC_"`).

**Comment by isaacfink** [comment]
@amyegan I had multiple iterations of the project amd at the moment its working so I'll leave it for now and get back here if it starts acting up again
